src/features/create-token/steps/DetailsForm.test.tsx

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+
+import userEvent from '@testing-library/user-event';
+import { fireEvent, render, waitFor, screen } from '@testing-library/react';
+
+import { ZUIProvider } from '@zero-tech/zui/ZUIProvider';
+import { DetailsForm, DetailsFormProps } from './';
+import { CreateTokenFormContext } from '../';
+
+let onSubmit = jest.fn();
+
+const DEFAULT_PROPS: DetailsFormProps = {
+	onClose: jest.fn(),
+};
+
+const DEFAULT_PROVIDER_VALUES = {
+	stepId: 'details',
+	title: 'Create Token',
+	details: {
+		mediaType: undefined,
+		previewUrl: '',
+		name: '',
+		symbol: '',
+	},
+	tokenomics: {
+		tokenCount: '',
+		initialTokenSupplyWalletAddress: '',
+		adminWalletAddress: '',
+	},
+	onStepUpdate: jest.fn(),
+	onTitleUpdate: jest.fn(),
+	onDetailsChange: jest.fn(),
+	onDetailsSubmit: onSubmit,
+	onTokenomicsSubmit: jest.fn(),
+	onLaunchSubmit: jest.fn(),
+};
+
+describe('<DetailsForm />', () => {
+	beforeEach(() => jest.resetAllMocks());
+
+	test('should correctly validate required name field', async () => {
+		render (
+			<ZUIProvider>
+				<CreateTokenFormContext.Provider value={DEFAULT_PROVIDER_VALUES}>
+					<DetailsForm {...DEFAULT_PROPS} />
+				</CreateTokenFormContext.Provider>
+			</ZUIProvider>
+		);
+
+		fireEvent.blur(screen.getByPlaceholderText(/Enter name.../i));
+		fireEvent.click(
+			screen.getByRole('button', {
+				name: 'Next',
+			}),
+		);
+
+		await waitFor(() =>
+			expect(screen.getByText('The name field is required.')).not.toBe(null),
```

Comment:
could we use `.not.toBeInTheDocument` over `.not.toBe(null)`. This will be consistent with other tests. I think Brett mentioned this in another PR so best to follow that.